driver-core/src/main/com/mongodb/BasicDBObject.java

```diff
@@ -58,7 +66,7 @@
      * @mongodb.driver.manual reference/mongodb-extended-json/ MongoDB Extended JSON
      */
     public static BasicDBObject parse(final String json) {
-        return parse(json, getDefaultCodecRegistry().get(BasicDBObject.class));
+        return parse(json, DBObjectCodec.getDefaultRegistry().get(BasicDBObject.class));
```

Comment:
Can't just use new DBObjectCodec() here since that's a Decoder<DBObject> rather than a Decoder<BasicDBObject>